__tests__/commands/install/integration.js

```diff
@@ -437,6 +437,24 @@ test.concurrent('install with file: protocol as default', (): Promise<void> => {
   });
 });
 
+test.concurrent('install file: dedupe dependencies', async (): Promise<void> => {
+  await runInstall({}, 'install-file-dedupe-dependencies', async (config, reporter) => {
+    expect(await fs.exists(path.join(config.cwd, 'node_modules', 'a', 'node_modules'))).toEqual(false);
+
+    // Add b as a dependency of c, as an absolute path
+    const pkgJsonContents = await fs.readFile(path.join(config.cwd, 'c/package.json'));
+    const pkgJson = JSON.parse(pkgJsonContents);
+    pkgJson.dependencies = {b: `file:${path.join(config.cwd, 'b')}`};
+    await fs.writeFile(path.join(config.cwd, 'c/package.json'), JSON.stringify(pkgJson));
```

Comment:
Why are we doing this? Feels like this needs to be a separate test case or handled via a `yarn add b@file:...` command instead of directly manipulating the `package.json` file in the fixture.